src/substrate/subscribeFunc.ts

```diff
@@ -104,24 +103,37 @@ export const subscribeEvents: SubscribeFunc<ApiPromise, Block, ISubscribeOptions
         && (!offlineRange || !offlineRange.startBlock || offlineRange.startBlock < lastBlockNumber)) {
       offlineRange = { startBlock: lastBlockNumber };
     }
+    // if running in archival mode and can't determine the offlineRange then start from Block 0
+    if(archival && (!offlineRange || !offlineRange.startBlock)){
+      offlineRange = {startBlock: 0}
+    }
 
     // if we can't figure out when the last block we saw was, do nothing
     // (i.e. don't try and fetch all events from block 0 onward)
-    if (!offlineRange || !offlineRange.startBlock) {
-      log.warn('Unable to determine offline time range.');
-      return;
-    }
-
-    // poll the missed blocks for events
-    try {
-      const blocks = await poller.poll(offlineRange);
-      await Promise.all(blocks.map(processBlockFn));
-    } catch (e) {
-      log.error(`Block polling failed after disconnect at block ${offlineRange.startBlock}`);
+    // if (!offlineRange || !offlineRange.startBlock) {
```

Comment:
This should be uncommented and switched on the "else" branch for archival mode. We want this behavior when running on CW.

Reply:
restructured, the script should execute normally if the archival flag is `false`